plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectJsonPushdownWithSplits.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3select;
+
+import io.airlift.units.DataSize;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.MaterializedResult;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static io.trino.plugin.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static io.trino.plugin.hive.HiveColumnHandle.createBaseColumn;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.newSession;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+import static org.testng.Assert.assertTrue;
+
+public class TestHiveFileSystemS3SelectJsonPushdownWithSplits
+{
+    private String host;
+    private int port;
+    private String databaseName;
+    private String awsAccessKey;
+    private String awsSecretKey;
+    private String writableBucket;
+    private String testDirectory;
+
+    private SchemaTableName tableJsonWithSplits;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.databaseName = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+
+        this.tableJsonWithSplits = new SchemaTableName(databaseName, "trino_s3select_test_json_scan_range_pushdown");
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{15, 10, 6, 12}, {50, 30, 2, 4}};
+    }
+
+    @Test(dataProvider = "testSplitSize")
+    public void testQueryPushdownWithSplitSizeForJson(int maxSplitSizeKB,
+                                                      int maxInitialSplitSizeKB,
+                                                      int minSplitCount,
+                                                      int maxSplitCount)
+    {
+        S3SelectTestHelper s3SelectTestHelper = null;
```

Comment:
Can the s3SelectTestHelper be created once per class instead of once per test?

Reply:
No, these tests can run in parallel, having this as a field might cause concurrency issues, as they would need to update the configuration in the same variable.